src/lib/Microsoft.Health.Events/EventCheckpointing/StorageCheckpointClient.cs

```diff
@@ -33,21 +34,30 @@ public StorageCheckpointClient(StorageCheckpointOptions options, ITelemetryLogge
         {
             EnsureArg.IsNotNull(options);
             EnsureArg.IsNotNullOrWhiteSpace(options.BlobPrefix);
-            EnsureArg.IsNotNullOrWhiteSpace(options.BlobStorageConnectionString);
-            EnsureArg.IsNotNullOrWhiteSpace(options.BlobContainerName);
             EnsureArg.IsNotNullOrWhiteSpace(options.CheckpointBatchCount);
 
+            var accountName = EnsureArg.IsNotNullOrWhiteSpace(options.BlobStorageAccountName);
+            var containerName = EnsureArg.IsNotNullOrWhiteSpace(options.BlobContainerName);
+
             BlobPrefix = options.BlobPrefix;
 
             _lastCheckpointMaxCount = int.Parse(options.CheckpointBatchCount);
             _checkpoints = new ConcurrentDictionary<string, Checkpoint>();
             _lastCheckpointTracker = new ConcurrentDictionary<string, int>();
-            _storageClient = new BlobContainerClient(options.BlobStorageConnectionString, options.BlobContainerName);
+
+            var checkpointUri = new Uri($"https://{accountName}.blob.core.windows.net/{containerName}");
+            var credential = new DefaultAzureCredential();
```

Comment:
I'm not sure if we should tackle this now (because of our short timeline), but I'm wondering if it makes sense to support both connection string and managed identity. When working with one of our customers, we found a case where the IT team could not deploy our the IoMT Connector ARM template that used MI. The reason was that the user did not have the required permissions in the AAD tenant.

Reply:
Ideally this should be DIed.  Can support default like here, connection string later if needed, or cert based identity if needed in the future.

---
In reply to: [565584072](https://github.com/microsoft/iomt-fhir/pull/87#discussion_r565584072) [](ancestors = 565584072)

Reply:
I've updated the code so that this is injected.